crates/ide_db/src/search.rs

```diff
@@ -55,22 +55,10 @@ impl IntoIterator for UsageSearchResult {
 #[derive(Debug, Clone)]
 pub struct FileReference {
     pub range: TextRange,
-    pub kind: ReferenceKind,
+    pub name: ast::NameLike,
```

Comment:
I wonder if that should be `pub name: InFile<ast::NameLike>`? That should handle macros in a principled way.

I also think that NameLike should probably be defined here. I imagine it might have a shape closer to NameRefClass, but for ast. For example, it can have a dedicated variant for shorthands.